{math}[foss/2019a] Octave 5.1.0 w/ updated dependencies

crubb opened the pull request:
This PR adds

- Octave-5.1.0-foss-2019a.eb

with updated dependencies:

- Bison-3.3.2-GCCcore-8.2.0.eb
- FLTK-1.3.5-foss-2019a.eb
- GL2PS-1.4.0-foss-2019a.eb
- GraphicsMagick-1.3.34-foss-2019a.eb
- Qhull-2019.1-GCCcore-8.2.0.eb
- qrupdate-1.1.2-foss-2019a.eb

boegel commented on easybuild/easyconfigs/f/FLTK/FLTK-1.3.5-foss-2019a.eb:
```diff
@@ -0,0 +1,41 @@
+ and MacOS X. FLTK provides modern GUI functionality without the bloat and supports 3D graphics via OpenGL
+ and its built-in GLUT emulation."""
+
+toolchain = {'name': 'foss', 'version': '2019a'}
```
@crubb Why not use the existing `FLTK-1.3.5-GCC-8.2.0-2.31.1.eb`?

crubb replied:
Didn't see it, sorry. Fixed!

boegel commented on easybuild/easyconfigs/q/qrupdate/qrupdate-1.1.2-foss-2019a.eb:
```diff
@@ -0,0 +1,31 @@
+homepage = 'https://sourceforge.net/projects/qrupdate/'
+description = """qrupdate is a Fortran library for fast updates of QR and Cholesky decompositions."""
+
+toolchain = {'name': 'foss', 'version': '2019a'}
```
@crubb Move to `GCC` toolchain (not `GCCcore`, since it's Fortran and compiling with `iccifort` is important if this is needed for something built with `intel` toolchain)

crubb replied:
Moved to `GCC/8.2.20-2.31.1`.

crubb commented:
I re-added the pkgconfig patch for Qhull (otherwise EasyBuild sanity checks would have failed), fixed the dependencies, as you suggested, and also added the `general` toolbox for Octave.

boegel commented:
Test report by @boegel
**FAILED**
Build succeeded for 12 out of 13 (6 easyconfigs in this PR)
generoso-1 - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/dd3ec3cf8dbdc7791916551bfa40ca9a for a full test report.

boegel commented:
Test report by @boegel
**FAILED**
Build succeeded for 11 out of 12 (6 easyconfigs in this PR)
node3101.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/8f99a2c4dec9f7732e1c15ad8377cce2 for a full test report.

boegel commented:
Test report by @boegel
**FAILED**
Build succeeded for 10 out of 11 (6 easyconfigs in this PR)
node2548.golett.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/37826986987e9b1491cde049e57dfb82 for a full test report.

boegel commented:
@crubb Hmm, looks like a missing dependency?

```
/software/Octave/5.1.0-foss-2019a/bin/octave-cli-5.1.0: symbol lookup error: /software/Octave/5.1.0-foss-2019a/lib/octave/5.1.0/liboctinterp.so.7: undefined symbol: jpeg_resync_to_restart
```

zao commented:
Test report by @zao
**FAILED**
Build succeeded for 27 out of 29 (6 easyconfigs in this PR)
freja - Linux ubuntu 18.04, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 2.7.17
See https://gist.github.com/6de59a6939e5c580f17cb2ca81f9e314 for a full test report.

zao commented:
Interesting, seems like `gnuplot-5.2.6-GCCcore-8.2.0.eb` has a sneaky dependency on Lua, based on my test report?

zao commented:
System has a whole bunch of Lua runtimes installed for Lmod, and `liblua5.[123]-dev`. Should the easyconfig explicitly disable Lua capabilities or add a dependency?

gnuplot config.log excerpts, where it fails to find Lua via pkg-config, but deduces that there's a bit of it present as it blindly manages to link a runtime library:
```
checking for LUA... no
checking for LUA... no
checking for LUA... no
checking for LUA... no
configure: WARNING: Could not find support for lua using pkg-config.
checking for library containing luaL_openlibs... -llua5.3
checking lua.h usability... no
checking lua.h presence... no
checking for lua.h... no
configure: WARNING: please add path to lua.h to CPPFLAGS in Makefile
```
```
  lua/TikZ terminal: yes
```

terjekv commented:
Test report by @terjekv
**FAILED**
Build succeeded for 27 out of 29 (6 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
See https://gist.github.com/d0b2d4f562f62731f514a39076b0ee3b for a full test report.

zao commented:
I see that my problem has been discovered by @branfosj before in #9773.

boegel commented:
@zao Please try again on top of latest `develop` now #9773 is merged?

@terjekv Is that just a temporary download problem for `qrupdate`?

terjekv commented:
Looks like a temporary error, restarting the build now.

terjekv commented:
Test report by @terjekv
**FAILED**
Build succeeded for 1 out of 2 (6 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
See https://gist.github.com/088d90ebb9a6e6e466394144496a180f for a full test report.

zao commented:
Test report by @zao
**FAILED**
Build succeeded for 0 out of 1 (6 easyconfigs in this PR)
freja - Linux ubuntu 18.04, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 2.7.17
See https://gist.github.com/086d8ef64d5fa9e371f38003d904fbb8 for a full test report.

terjekv commented:
So the crash is the jpeg issue, so that's okay, but I'm also concerned about the disabling of GUI. Is that expected?
```
== 2020-03-03 11:01:29,752 build_log.py:169 ERROR EasyBuild crashed with an error (at easybuild/base/exceptions.py:124 in __init__): cmd "octave --eval 'pkg prefix /tmp/terjekv/tmp-installs/9680/software/Octave/5.1.0-foss-2019a/share/octave/packages /tmp/terjekv/tmp-installs/9680/software/Octave/5.1.0-foss-2019a/share/octave/packages-arch-dep; pkg install -global /opt/uio/modules/rhel8/easybuild/sources/o/Octave/extensions/control-3.2.0.tar.gz'" exited with exit code 127 and output:
octave: X11 DISPLAY environment variable not set
octave: disabling GUI features
/tmp/terjekv/tmp-installs/9680/software/Octave/5.1.0-foss-2019a/bin/octave-cli-5.1.0: symbol lookup error: /tmp/terjekv/tmp-installs/9680/software/Octave/5.1.0-foss-2019a/lib/octave/5.1.0/liboctinterp.so.7: undefined symbol: jpeg_resync_to_restart
 (at easybuild/tools/run.py:533 in parse_cmd_output)
== 2020-03-03 11:01:29,752 easyblock.py:3142 WARNING build failed (first 300 chars): cmd "octave --eval 'pkg prefix /tmp/terjekv/tmp-installs/9680/software/Octave/5.1.0-foss-2019a/share/octave/packages /tmp/terjekv/tmp-installs/9680/software/Octave/5.1.0-foss-2019a/share/octave/packages-arch-dep; pkg install -global /opt/uio/modules/rhel8/easybuild/sources/o/Octave/extensions/contro
== 2020-03-03 11:01:29,752 easyblock.py:295 INFO Closing log for application name Octave version 5.1.0
```

zao commented:
My test failure seems to be due to GraphicsMagick finding WEBP libraries on the system to link to, implicitly enabling their `--with-webp=yes` feature.

GraphicsMagick seems to install as static libraries with corresponding `.la` files that mention the dependency libraries it needs, while the pkg-config files do not mention any dependencies at all. It's possible that the Octave linking process misses some of these transitive dependencies.

zao commented:
Test report by @zao
**FAILED**
Build succeeded for 7 out of 8 (6 easyconfigs in this PR)
freja - Linux ubuntu 18.04, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 2.7.17
See https://gist.github.com/3ae68f264179f33afe18b522e073bcc2 for a full test report.

zao commented:
@crubb I recommend adding `configopts = ' --enable-shared'` to GraphicsMagick, seems to pass the Octave tests with it on my previously troubled machine.

crubb commented:
@zao @terjekv Thanks for all the testing and the suggestion, @zao! Could you re-run the tests?

Thanks!

zao commented:
Test report by @zao
**FAILED**
Build succeeded for 7 out of 8 (7 easyconfigs in this PR)
freja - Linux ubuntu 18.04, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 2.7.17
See https://gist.github.com/0d36d96e61a45ff1cc75e04c59af92a3 for a full test report.

terjekv commented:
Test report by @terjekv
**FAILED**
Build succeeded for 37 out of 39 (7 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
See https://gist.github.com/1b5f4f4e1a58235fed0a3a0227ca7b81 for a full test report.

crubb commented:
.... added the `--enable-shared` to the correct `.eb` now and cleaned up the wrongly edited file - sorry!

zao commented:
Test report by @zao
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
freja - Linux ubuntu 18.04, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 2.7.17
See https://gist.github.com/a8d5d5e69a67e5d7623fe8f5fc0041b3 for a full test report.

zao commented:
Going in, thanks @crubb!

boegel commented:
The `GraphisMagick` easyconfig included here failed to install for me because `webp/mux.h` was missing... Fixed with #10274
